src/Features/OnboardingTasks/Task.php

```diff
@@ -0,0 +1,296 @@
+	 *
+	 * @var string
+	 */
+	protected $duration_to_ms = array(
+		'day'  => DAY_IN_SECONDS * 1000,
+		'hour' => HOUR_IN_SECONDS * 1000,
+		'week' => WEEK_IN_SECONDS * 1000,
+	);
+
+	/**
+	 * Constructor
+	 *
+	 * @param array $data Task list data.
+	 */
+	public function __construct( $data = array() ) {
+		$defaults = array(
+			'id'             => null,
+			'title'          => '',
+			'content'        => '',
+			'action_label'   => __( "Let's go", 'woocommerce-admin' ),
+			'action_url'     => null,
+			'is_complete'    => false,
+			'can_view'       => true,
+			'time'           => null,
+			'is_dismissable' => false,
+			'is_snoozeable'  => false,
+			'snoozed_until'  => null,
+		);
+
+		$data = wp_parse_args( $data, $defaults );
+
+		$this->id             = (string) $data['id'];
+		$this->title          = (string) $data['title'];
+		$this->content        = (string) $data['content'];
+		$this->action_label   = (string) $data['action_label'];
+		$this->action_url     = (string) $data['action_url'];
+		$this->is_complete    = (bool) $data['is_complete'];
+		$this->can_view       = (bool) $data['can_view'];
+		$this->time           = (string) $data['time'];
+		$this->is_dismissable = (bool) $data['is_dismissable'];
+		$this->is_snoozeable  = (bool) $data['is_snoozeable'];
+
+		$snoozed_tasks = get_option( self::SNOOZED_OPTION, array() );
+		if ( isset( $snoozed_tasks[ $this->id ] ) ) {
+			$this->snoozed_until = $snoozed_tasks[ $this->id ];
+		}
+	}
+
+	/**
+	 * Bool for task dismissal.
+	 *
+	 * @return bool
+	 */
+	public function is_dismissed() {
+		if ( ! $this->is_dismissable ) {
+			return false;
+		}
+
+		$dismissed = get_option( self::DISMISSED_OPTION, array() );
+
+		return in_array( $this->id, $dismissed, true );
+	}
+
+	/**
+	 * Dismiss the task.
+	 *
+	 * @return bool
+	 */
+	public function dismiss() {
+		if ( ! $this->is_dismissable ) {
+			return false;
+		}
+
+		$dismissed   = get_option( self::DISMISSED_OPTION, array() );
+		$dismissed[] = $this->id;
+		$update      = update_option( self::DISMISSED_OPTION, array_unique( $dismissed ) );
+
+		if ( $update ) {
+			wc_admin_record_tracks_event( 'tasklist_dismiss_task', array( 'task_name' => $this->id ) );
+		}
+
+		return $update;
+	}
+
+	/**
+	 * Undo task dismissal.
+	 *
+	 * @return bool
+	 */
+	public function undo_dismiss() {
+		$dismissed = get_option( self::DISMISSED_OPTION, array() );
+		$dismissed = array_diff( $dismissed, array( $this->id ) );
+		$update    = update_option( self::DISMISSED_OPTION, $dismissed );
+
+		if ( $update ) {
+			wc_admin_record_tracks_event( 'tasklist_undo_dismiss_task', array( 'task_name' => $this->id ) );
+		}
+
+		return $update;
+	}
+
+	/**
+	 * Bool for task snoozed.
+	 *
+	 * @return bool
+	 */
+	public function is_snoozed() {
+		if ( ! $this->is_snoozeable ) {
+			return false;
+		}
+
+		$snoozed = get_option( self::SNOOZED_OPTION, array() );
+
+		return isset( $snoozed[ $this->id ] ) && $snoozed > ( time() * 1000 );
```

Comment:
I think you need to target the ID here?
```suggestion
		return isset( $snoozed[ $this->id ] ) && $snoozed[ $this->id ] > ( time() * 1000 );
```

Reply:
Great catch, @joelclimbsthings!  Updated in https://github.com/woocommerce/woocommerce-admin/pull/7665/commits/ccf27f4ca45e7dbf1174e20df6dd52c9b8318611 and added tests around the time.